sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -102,7 +102,12 @@ internal void AddPolicies(RequestContext? context)
         public bool TryGetProperty(string name, out object? value)
         {
             value = null;
-            return _properties?.TryGetValue(name, out value) == true;
+            if (_typeProperties == null || !_typeProperties.TryGetValue(typeof(MessagePropertyKey), out var rawValue))
+            {
+                return false;
+            }
+            var properties = (Dictionary<string, object>)rawValue!;
```

Comment:
should this be an `as` cast instead followed by a null check in case the type that is in there is not actually that dictionary?